pallets/parachain-staking/src/lib.rs

```diff
@@ -2774,66 +2783,148 @@ pub mod pallet {
 
+			// if there's no previous round, i.e, genesis round, then skip
+			reads = reads.saturating_add(Weight::from_parts(1_u64, 0));
+			if Self::round().current.is_zero() {
+				return
+			}
+
+			if let Some(payout_info) = DelayedPayoutInfo::<T>::get() {
+				if let Some((author, block_num)) =
+					CollatorBlocks::<T>::iter_prefix(payout_info.round).drain().next()
+				{
+					let pot = Self::account_id();
+					// get collator's staking info
+					if let Some(state) = AtStake::<T>::take(payout_info.round, author) {
+						// calculate reward for collator from previous round
+						let now_reward = Self::get_collator_reward_per_session(
+							&state,
+							block_num,
+							payout_info.total_stake,
+							payout_info.total_issuance,
+						);
+						Self::do_reward(&pot, &now_reward.owner, now_reward.amount);
+						reads = reads.saturating_add(Weight::from_parts(1_u64, 0));
+						writes = writes.saturating_add(Weight::from_parts(1_u64, 0));
+
+						// calculate reward for collator's delegates from previous round
+						let now_rewards = Self::get_delgators_reward_per_session(
+							&state,
+							block_num,
+							payout_info.total_stake,
+							payout_info.total_issuance,
+						);
+
+						let len = now_rewards.len().saturated_into::<u64>();
+						now_rewards.into_iter().for_each(|x| {
+							Self::do_reward(&pot, &x.owner, x.amount);
+						});
+						reads = reads.saturating_add(Weight::from_parts(len, 0));
+						writes = writes.saturating_add(Weight::from_parts(len, 0));
+					}
+				} else {
+					// Kill storage
+					DelayedPayoutInfo::<T>::kill();
+
+					// If there were no more bock authors left, we should clean up shapshot of
+					// remaining collators that didn't author blocks
+					// we do this in the block after the last payout is done to reduce computational
+					// cost for block with last payout
+					let cursor = AtStake::<T>::clear_prefix(payout_info.round, u32::MAX, None);
+					if cursor.maybe_cursor.is_none() {
+						log::debug!("snapshot cleared for round {:?}", payout_info.round);
+					} else {
+						// This is an ambiguous case
+						// We cannot just iterate till maybe_cursor is none, as each time the time
+						// complexity is O(n)
+						log::error!(
+							"snapshot not entirely cleared for round {:?}",
+							payout_info.round
+						);
+					}
+				}
+			}
+			frame_system::Pallet::<T>::register_extra_weight_unchecked(
+				T::DbWeight::get().reads_writes(reads.ref_time(), writes.ref_time()),
+				DispatchClass::Mandatory,
+			);
+		}
+
+		pub(crate) fn pot_issuance() -> (Weight, BalanceOf<T>) {
 			let pot = Self::account_id();
-			let ed = <T::Currency as frame_support::traits::fungible::Inspect<T::AccountId>>::minimum_balance();
-			let issue_number = if ed == T::CurrencyBalance::from(0_u32) {
-				T::Currency::reducible_balance(&pot, Preservation::Preserve, Fortitude::Polite)
-					// Avoid the pot complaint no balance there
-					.checked_sub(&T::CurrencyBalance::from(10_u32))
```

Comment:
Please keep this logic. Otherwise, every time after we distribute the reward, one event will be emitted because we'll drain all the tokens in the pot

Reply:
Incorporated this back in🤝